StartBonusActivation/handler.ts

```diff
@@ -386,69 +352,55 @@ export function StartBonusActivationHandler(
           familyUID
         }))
       )
-      .chain<BonusActivationWithFamilyUID>(({ dsu, familyUID }) => {
-        // this part is on a sub-chain as it handles the lock/unlock protection mechanism
-        // familyUID serves as lock context and thus is needed in scope for every sub-task
-        return (
-          taskEither
-            .of<StartBonusActivationResponse, void>(void 0)
-
-            .chain(_ =>
-              createBonusActivation(
-                bonusActivationModel,
-                fiscalCode,
-                familyUID,
-                dsu
-              )
-            )
-            .chain(bonusActivation =>
-              runStartBonusActivationOrchestrator(
-                client,
-                bonusActivation,
-                fiscalCode
-              ).map(_ => bonusActivation)
-            )
-            // the following is basically:
-            // on right, just pass it
-            // on left, perform unlock but then pass the original left value
-            .foldTaskEither(
-              l => {
-                defaultClient.trackException({
-                  exception: new Error(l.detail),
-                  properties: {
-                    name: "bonus.activation.start"
-                  }
-                });
-                return relaseLockForUserFamily(
-                  bonusLeaseModel,
-                  familyUID
-                ).foldTaskEither(
-                  _ => fromEither(left(l)),
-                  _ => fromEither(left(l))
-                );
-              },
-              r => fromEither(right(r))
-            )
-        );
-      })
+      .chain<BonusActivationWithFamilyUID>(({ dsu, familyUID }) =>
+        createBonusActivation(bonusActivationModel, fiscalCode, familyUID, dsu)
+          .map(bonusActivation => {
+            // Send the (bonusId, applicantFiscalCode) to the bonus activations queue
+            // in order to be processed later (asynchronously)
+            // tslint:disable-next-line: no-object-mutation
+            context.bindings.bonusActivation = ContinueBonusActivationInput.encode(
+              {
+                applicantFiscalCode: bonusActivation.applicantFiscalCode,
+                bonusId: bonusActivation.id
+              }
+            );
```

Comment:
As this introduces a side effect, can we push it at the end of the pipeline? Maybe [in the final fold](https://github.com/pagopa/io-functions-bonus/pull/90/files#diff-34aa739be32c08eaf7ea3d2629f4a9cdR399), where the positive response is prepared.

Reply:
very good suggestion, applied!